src/Test.UnitTests.Sarif.Multitool/ResultMatchingCommandTests.cs

```diff
@@ -0,0 +1,172 @@
+﻿using FluentAssertions;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Baseline.ResultMatching;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Multitool
+{
+    public class ResultMatchingCommandTests
+    {
+        [Fact]
+        public void ResultMatchingCommand_TestingSameResult()
+        {
+            SarifLog baselineSarif, currentSarif;
+
+            #region .: Arrange :.
+
+            baselineSarif = CreateBaseline();
+            currentSarif = CreateBaseline();
+
+            #endregion
+
+            ISarifLogMatcher matcher = ResultMatchingBaselinerFactory.GetDefaultResultMatchingBaseliner();
+            SarifLog output = matcher.Match(new SarifLog[] { baselineSarif }, new SarifLog[] { currentSarif }).First();
+
+            output.Runs.First().Results.First().BaselineState.Should().Be(BaselineState.Unchanged);
+        }
+
+        [Fact]
+        public void ResultMatchingCommand_TestingNewResults()
+        {
+            SarifLog baselineSarif, currentSarif;
+
+            #region .: Arrange :.
+
+            baselineSarif = CreateBaseline();
+            currentSarif = CreateCurrent();
+
+            #endregion
+
+            ISarifLogMatcher matcher = ResultMatchingBaselinerFactory.GetDefaultResultMatchingBaseliner();
+            SarifLog output = matcher.Match(new SarifLog[] { baselineSarif }, new SarifLog[] { currentSarif }).First();
+
+            output.Runs[0].Results[0].BaselineState.Should().Be(BaselineState.Unchanged);
```

Comment:
>Unchanged [](start = 78, length = 9)

This test doesn't actually appear to be verifying the 'new' state. But! As mentioned, I think the other test content does, just noting this for completeness. :)

Reply:
just added more tests. Now we have absent/unchanged/new

---
In reply to: [502494403](https://github.com/microsoft/sarif-sdk/pull/2102#discussion_r502494403) [](ancestors = 502494403)